gridfs/grid_file.py

```diff
@@ -179,7 +179,7 @@ def __create_index(self, collection, index, unique):
                 indexes = list(collection.list_indexes())
             except OperationFailure:
                 indexes = []
-            if index not in indexes:
+            if dict(index) not in [idx['key'].to_dict() for idx in indexes]:
```

Comment:
According to SON object's compare strategy ([\_\_eq\_\_](https://github.com/mongodb/mongo-python-driver/blob/master/bson/son.py#L206)), it should ensure items's order.
```
In [59]: SON([(u'filename', 1.0), (u'uploadDate', 1.0)]) == SON([(u'uploadDate', 1.0), (u'filename', 1.0)])
Out[59]: False
```
But here we don't know the order returned from `list_indexes()`, so I use dict to compare.